tfx/examples/imdb_lstm/imdb_pipeline_native_keras.py

```diff
@@ -0,0 +1,180 @@
+
+from tfx.proto import pusher_pb2
+from tfx.proto import trainer_pb2 
+
+from tfx.orchestration import metadata
+from tfx.orchestration import pipeline 
+from tfx.orchestration.beam.beam_dag_runner import BeamDagRunner
+
+_pipeline_name = 'imdb_lstm_native_keras'
+
+# This exmaple assumes the utility function is in ~/imdb
+_imdb_root = os.path.join(os.environ['HOME'], 'imdb_lstm')
+_data_root = os.path.join(_imdb_root, 'data') 
+# Python module file to inject customized logic into the TFX components. The
+# Transform and Trainer both require user-defined functions to run successfully.
+_module_file = os.path.join(_imdb_root, 'imdb_utils_native_keras.py')
+# Path which can be listened to by the model server.  Pusher will output the
+# trained model here.
+_serving_model_dir = os.path.join(_imdb_root, 'serving_model', _pipeline_name)
+
+# Directory and data locations.  This example assumes all of the flowers
+# example code and metadata library is relative to $HOME, but you can store
+# these files anywhere on your local filesystem.
+_tfx_root = os.path.join(os.environ['HOME'], 'tfx')
+_pipeline_root = os.path.join(_tfx_root, 'pipelines', _pipeline_name)
+# Sqlite ML-metadata db path.
+_metadata_path = os.path.join(_tfx_root, 'metadata', _pipeline_name,
+                              'metadata.db')
+
+def _create_pipeline(pipeline_name: Text, pipeline_root: Text, data_root: Text,
+                     module_file: Text, serving_model_dir: Text,
+                     metadata_path: Text,
+                     direct_num_workers: int) -> pipeline.Pipeline:
+    """Implements the imdb sentiment analysis pipline with TFX."""
+    examples = external_input(data_root)
+    # Brings data in to the pipline
+    example_gen = CsvExampleGen(input=examples)
+
+    # Computes statistics over data for visualization and example validation.
+    statistics_gen = StatisticsGen(examples=example_gen.outputs['examples'])
+
+    # Generates schema based on statistics files.
+    schema_gen = SchemaGen(
+      statistics=statistics_gen.outputs['statistics'], infer_feature_shape=True)
+
+    # Performs anomaly detection based on statistics and data schema.
+    example_validator = ExampleValidator(
+      statistics=statistics_gen.outputs['statistics'],
+      schema=schema_gen.outputs['schema'])
+
+    # Performs transformations and feature engineering in training and serving.
+    transform = Transform(
+      examples=example_gen.outputs['examples'],
+      schema=schema_gen.outputs['schema'],
+      module_file=module_file)
+
+     # Uses user-provided Python function that trains a model using TF-Learn.
+    trainer = Trainer(
+      module_file=module_file,
+      custom_executor_spec=executor_spec.ExecutorClassSpec(GenericExecutor),
+      examples=transform.outputs['transformed_examples'],
+      transform_graph=transform.outputs['transform_graph'],
+      schema=schema_gen.outputs['schema'],
+      train_args=trainer_pb2.TrainArgs(),
```

Comment:
We should probably set these with num_steps like so: https://www.tensorflow.org/tfx/guide/trainer#configuring_a_trainer_component

Reply:
i think the issue here is, when constructing a tfx example, we assume a big dataset. Potentially we don't know the full size.
Yet in this case, the #of steps is in one epoch is #traindata / batch_size
so im not sure which is the better solution